src/Spice86.Core/Emulator/Devices/Input/Mouse/Mouse.cs

```diff
@@ -0,0 +1,176 @@
+    private const int IrqNumber = 12;
+    private readonly IGui? _gui;
+    private readonly ILoggerService _logger;
+    private long _lastUpdateTimestamp;
+    private bool _previousIsLeftButtonDown;
+    private bool _previousIsMiddleButtonDown;
+    private bool _previousIsRightButtonDown;
+    private double _previousMouseXRelative;
+    private double _previousMouseYRelative;
+    private int _sampleRate = 100;
+    private long _sampleRateTicks;
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="Mouse" /> class.
+    /// </summary>
+    /// <param name="machine">The emulator machine.</param>
+    /// <param name="gui">The graphical user interface. Is null in headless mode.</param>
+    /// <param name="configuration">to get the mouse type from</param>
+    /// <param name="loggerService">The logger service implementation.</param>
+    public Mouse(Machine machine, IGui? gui, Configuration configuration, ILoggerService loggerService) : base(machine, configuration, loggerService) {
+        _gui = gui;
+        MouseType = configuration.Mouse;
+        _logger = loggerService;
+        _sampleRateTicks = TimeSpan.TicksPerSecond / _sampleRate;
+        Initialize();
+    }
+
+    /// <inheritdoc />
+    public double DeltaY { get; private set; }
+
+    /// <inheritdoc />
+    public double DeltaX { get; private set; }
+
+    /// <inheritdoc />
+    public int ButtonCount => 3;
+
+    /// <inheritdoc />
+    public double MouseXRelative { get; set; }
+
+    /// <inheritdoc />
+    public double MouseYRelative { get; set; }
+
+    /// <inheritdoc />
+    public MouseEventMask LastTrigger { get; private set; }
+
+    /// <inheritdoc />
+    public int SampleRate {
+        get => _sampleRate;
+        set {
+            _sampleRate = value;
+            _sampleRateTicks = TimeSpan.TicksPerSecond / _sampleRate;
+        }
+    }
+
+    /// <inheritdoc />
+    public MouseType MouseType { get; }
+
+    /// <inheritdoc />
+    public bool IsLeftButtonDown { get; private set; }
+
+    /// <inheritdoc />
+    public bool IsRightButtonDown { get; private set; }
+
+    /// <inheritdoc />
+    public bool IsMiddleButtonDown { get; private set; }
+
+    /// <inheritdoc />
+    public int DoubleSpeedThreshold { get; set; }
+
+    /// <inheritdoc />
+    public int HorizontalMickeysPerPixel { get; set; } = 8;
+
+    /// <inheritdoc />
+    public int VerticalMickeysPerPixel { get; set; } = 16;
+
+    private void Initialize() {
+        if (_gui is not null && MouseType != MouseType.None) {
+            _gui.MouseButtonUp += OnMouseClick;
+            _gui.MouseButtonDown += OnMouseClick;
+            _gui.MouseMoved += OnMouseMoved;
+        }
+        if (_logger.IsEnabled(LogEventLevel.Information)) {
+            _logger.Information("Mouse initialized: {MouseType}", MouseType);
+        }
+    }
+
+    private void OnMouseMoved(object? sender, MouseMoveEventArgs eventArgs) {
```

Comment:
The sender argument is unused and could be removed.

Reply:
I'm using the standard .Net eventhandler delegate, which has that signature: 
".NET provides the [EventHandler](https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler) and [EventHandler&lt;TEventArgs>](https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler-1) delegates to support most event scenarios. Use the [EventHandler](https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler) delegate for all events that don't include event data. Use the [EventHandler&lt;TEventArgs>](https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler-1) delegate for events that include data about the event. These delegates have no return type value and take two parameters (an object for the source of the event and an object for event data)."